Add file name to file exceeds size limit error

ADustyOldMuffin opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->

Closes #10242 

## Description

- This will put the offending file name in the error message 

### Screenshots

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->

![image](https://user-images.githubusercontent.com/41391321/123903171-c8c75c00-d933-11eb-9a29-0bce9113bd3f.png)

## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes: Added offending file name to the file exceeds size limit error.

ADustyOldMuffin committed:
First stab at getting file in error message

ADustyOldMuffin committed:
lint fixes

ADustyOldMuffin commented:
So this is just a first stab (and IMO a kind of ugly one) but I was hoping to get feedback and maybe some direction on if this could be handled better.

I couldn't find a better place to get the offending file's name as the only place it was located was in the error messages returned.

I also know that this will not work with multiple files (see screenshot), but I wanted to wait on feedback before bothering with it since there might be an easier way to do it.

![image](https://user-images.githubusercontent.com/41391321/123903641-b13ca300-d934-11eb-86c9-3a07910e6f38.png)

sergiou87 commented on app/src/lib/helpers/regex.ts:
```diff
@@ -51,3 +51,19 @@ export function getMatches(text: string, re: RegExp): Array<RegExpExecArray> {
   return expression.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&')
 }
+
+export function getFileFromExceedsError(error: string): string | null {
```
Could you add an example of error message in the comments of this function?

Also I think it'd be better if this returned a list of files, instead of an already formatted string.

ADustyOldMuffin replied:
I added an example of the return, and also changed it to return an array of filenames

sergiou87 commented on app/src/lib/helpers/regex.ts:
```diff
@@ -51,3 +51,19 @@ export function getMatches(text: string, re: RegExp): Array<RegExpExecArray> {
+  const beginMatch = beginRegex.exec(error)
+  const endMatch = endRegex.exec(error)
+
+  if (beginMatch && beginMatch.index && endMatch && endMatch.index) {
```
FYI nowadays we prefer specific checks for `null` / `undefined`:
```suggestion
  if (beginMatch !== null && beginMatch.index && endMatch !== null && endMatch.index) {
```

What are the `.index` checks for exactly? 🤔

ADustyOldMuffin replied:
I just really like conditional statements... 

I was using `matches` before instead of `exec` in which the index might be undefined. I'll update it while I make the other changes.

sergiou87 reviewed (changes requested):
I like the idea! I think there is another error or something that just shows a list of files below the message, but I can't find it 🤔

ADustyOldMuffin committed:
Return an array of files and print out instead of a single

ADustyOldMuffin committed:
Update the error styling and added some comments

ADustyOldMuffin commented:
Current result, I tried to add some styling such as bold Files: but since it's just text I didn't have any luck. I use just newlines and tabs for moving the text around.

![image](https://user-images.githubusercontent.com/41391321/124003683-d8c75600-d99c-11eb-982c-e35d7340a85a.png)

ADustyOldMuffin committed:
I'll get linting right eventually

sergiou87 commented on app/src/lib/git/core.ts:
```diff
@@ -230,6 +231,17 @@ export async function git(
 
   log.error(errorMessage.join('\n'))
 
+  if (gitError && gitError === DugiteError.PushWithFileSizeExceedingLimit) {
```
You don't really need the `gitError && ` part here, right?
```suggestion
  if (gitError === DugiteError.PushWithFileSizeExceedingLimit) {
```

sergiou87 commented on app/src/lib/git/core.ts:
```diff
@@ -230,6 +231,17 @@ export async function git(
+      if (files && files !== '') {
+        gitResult.gitErrorDescription += '\n\nFiles:\n\t\t' + files
+      }
+    }
```
Since `result` is always an array, this can probably be simplified like this:
```suggestion
    const files = result.join('\n\t\t')

    if (files !== '') {
      gitResult.gitErrorDescription += '\n\nFiles:\n\t\t' + files
    }
```

sergiou87 reviewed (changes requested):
Thanks for addressing those issues! I have other 2 suggestions (sorry 😳 ).

Just food for thought (**not a requirement**), but I think it could be nice to also extract the size of those files from the error message, and show it like `MyBigFile.txt (235.00MB)`

sergiou87 committed:
Bump desktop-trampoline to 0.9.8 to include ssh-wrapper

sergiou87 committed:
Copy ssh-wrapper in macOS builds

sergiou87 committed:
Use ssh-wrapper on macOS as ssh command

sergiou87 committed:
Enable ssh askpass feature on macOS for beta

sergiou87 committed:
Remove all traces of the old askpass trampoline script :fire:  :tada:

sergiou87 committed:
Merge pull request #12765 from desktop/set-almost-immediate

Workaround for Electron issue with setImmediate

sergiou87 committed:
Bump version and changelog

sergiou87 committed:
Set DISPLAY env variable on Windows

sergiou87 committed:
Removed SSH improvements from release notes :sweat:

tidy-dev committed:
Merge pull request #12789 from desktop/fix-ssh-askpas-windows

Set DISPLAY env variable on Windows

billygriffin committed:
Update changelog.json

sergiou87 committed:
Merge pull request #12788 from desktop/releases/2.9.1-beta3

Release 2.9.1-beta3

tidy-dev committed:
Release 2.9.1 beta4

tidy-dev committed:
Merge pull request #12790 from desktop/releases/2.9.1-beta4

Release 2.9.1 beta4

tidy-dev committed:
Update warn-force-push-dialog.tsx

sergiou87 committed:
Cleanup

sergiou87 committed:
Only use ssh-wrapper on macOS DEV builds

sergiou87 committed:
Await for `envForRemoteOperation`

sergiou87 committed:
Merge pull request #12798 from desktop/await-for-env-variables

Await for `envForRemoteOperation`

sergiou87 committed:
Bump version and changelog to 2.9.1-beta5

sergiou87 committed:
Don't copy `ssh-wrapper` on non-dev builds

sergiou87 committed:
Store trampoline token in trampoline commands

sergiou87 committed:
Don't store SSH key passphrases directly, just send them back to the trampoline handler

tidy-dev committed:
Merge pull request #12792 from desktop/Remove-random-apostrophe

Update warn-force-push-dialog.tsx

sergiou87 committed:
Small refactor to store SSH key passphrases after a git command succeeds

sergiou87 committed:
Merge pull request #12782 from desktop/ssh-askpass-macos

Enable SSH improvements on macOS

sergiou87 committed:
Merge pull request #12800 from desktop/ssh-passphrase-improvements

SSH passphrase fixes and improvements

sergiou87 committed:
Merge pull request #12799 from desktop/releases/2.9.1-beta5

Release 2.9.1-beta5

sergiou87 committed:
Merge pull request #12783 from desktop/remove-old-askpass

Remove old askpass trampoline

sergiou87 committed:
Bump version and changelog to 2.9.1-beta6

sergiou87 committed:
Merge pull request #12802 from desktop/releases/2.9.1-beta6

Release 2.9.1-beta6

tidy-dev committed:
Merge branch 'development' into high-contrast

sergiou87 committed:
Remove pending SSH Key passphrases when needed

sergiou87 committed:
Merge pull request #12804 from desktop/more-ssh-passphrase-fixes

Remove pending SSH Key passphrases when users don't want to remember them

sergiou87 committed:
Bump version and changelog to 2.9.1-beta7

sergiou87 committed:
Merge pull request #12805 from desktop/releases/2.9.1-beta7

Release 2.9.1-beta7

tidy-dev committed:
Cleanup per pr comments

Co-Authored-By: Sergio Padrino <1083228+sergiou87@users.noreply.github.com>

tidy-dev committed:
Merge branch 'development' into rebase-multi-comit-op

tidy-dev committed:
Merge branch 'development' into high-contrast

tidy-dev committed:
Need to know what kind they are

sergiou87 commented:
> I think there is another error or something that just shows a list of files below the message, but I can't find it 🤔

@ADustyOldMuffin I found it! It's the confirmation dialog that shows up when you select multiple files in the `Changes` tab, right-click them and select `Discard changes`

![image](https://user-images.githubusercontent.com/1083228/130414581-edf03945-1f7a-490d-93d8-866697b7d02c.png)

You will find it in `discard-changes-dialog.tsx`, and key is in the `PathText` component: https://github.com/desktop/desktop/blob/f09e9ec53e771aa90787ff8e8cb058bf6437f5bc/app/src/ui/discard-changes/discard-changes-dialog.tsx/#L128-L151

sergiou87 committed:
Hook onUndo handler for cherry-picking

tidy-dev committed:
Merge pull request #12817 from desktop/fix-cherry-pick-undo

Fix `Undo` link not showing up for cherry-picks

tidy-dev committed:
Remove any to have better typing

tidy-dev committed:
Merge pull request #12819 from desktop/remove-any

Remove any to have better typing

tidy-dev committed:
Merge branch 'development' into high-contrast

sergiou87 committed:
Bump version and changelog to 2.9.1

sergiou87 committed:
Enable feature flags for 2.9.1

sergiou87 committed:
Apply suggestions from code review

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

sergiou87 committed:
Sort changelog lines for 2.9.1

Commit:
Fix order

Commit:
Add Dart

sergiou87 committed:
Merge pull request #12823 from desktop/releases/2.9.1

Release 2.9.1

tidy-dev committed:
Merge branch 'development' into high-contrast

tidy-dev committed:
Use colors suggested by consultant

tidy-dev committed:
Popup doesn't get cutoff

tidy-dev committed:
Foldouts should have borders

tidy-dev committed:
Make horizontal tab borders selection have more contrast

tidy-dev committed:
Fix where text disappears on hover for commits

tidy-dev committed:
Fix vertical menu item movement in preferences

tidy-dev committed:
Make it so user can choose a light background and still see syntax

tidy-dev committed:
Fix (or vastly improve) performance issue from dynamic diff gutter line width application

tidy-dev committed:
Fix (or vastly improve) performance issue from dynamic diff gutter line width application

tidy-dev committed:
Merge pull request #12834 from desktop/fix-line-number-perf-issue

Fix (or vastly improve) performance issue /w dynamic gutter lines

tidy-dev committed:
Release 2.9.2

tidy-dev committed:
Merge branch 'development' into rebase-multi-comit-op

sergiou87 committed:
Merge pull request #12835 from desktop/releases/2.9.2

Releases 2.9.2

sergiou87 committed:
Merge pull request #12827 from say25/add-dart

Add Syntax Highlighting For Dart

tidy-dev committed:
Fix Rebase Succes Banner

tidy-dev committed:
Merge pull request #12785 from desktop/rebase-multi-comit-op

Refactor rebase to be a multi-commit-operation

tidy-dev committed:
Silently error when no conflicts present for dismissing conflict dialog

tidy-dev committed:
Release 2.9.3-beta1

billygriffin committed:
Update changelog.json

tidy-dev committed:
Merge pull request #12840 from desktop/releases/2.9.3-beta1

Release 2.9.3-beta1

tsvetilian-ty committed:
Add minor version support for JetBrains IDEs

sergiou87 committed:
Fix Notepad++ and RStudio integration on Windows

sergiou87 committed:
Merge pull request #12849 from tsvetilian-ty/jetbrains-minor-version-detection

Add minor version support for JetBrains IDEs

sergiou87 committed:
Merge pull request #12849 from tsvetilian-ty/jetbrains-minor-version-detection

Add minor version support for JetBrains IDEs

tidy-dev committed:
Remove S3 bucket refs

sergiou87 committed:
Merge pull request #12863 from desktop/fix-win32-external-editors

Fix Notepad++ and RStudio integration on Windows

sergiou87 committed:
Merge pull request #12863 from desktop/fix-win32-external-editors

Fix Notepad++ and RStudio integration on Windows

tidy-dev committed:
Only on beta

tidy-dev committed:
Clarifying documentation

tidy-dev committed:
Move style building script into lib file - unit tests

niik committed:
Get rid of [Object object] strings

niik committed:
Don't log if the editor isn't installed on the system

tidy-dev committed:
Merge pull request #12866 from desktop/Remove-S3-bucket-refs

Remove S3 bucket refs

tidy-dev committed:
Merge pull request #12867 from desktop/reduce-multi-commit-error-noise

Silently error when no conflicts present for dismissing conflict dialog

sergiou87 committed:
Leverage TS to prevent mistakes supporting new editors on Windows

sergiou87 committed:
Get rid of the "expected installation checker"

niik committed:
Merge pull request #12872 from desktop/why-so-noisy-friend

Tidy up log messages around scanning for installed editors

sergiou87 committed:
Bump version and changelog to 2.9.3-beta2

sergiou87 committed:
Merge pull request #12876 from desktop/releases/2.9.3-beta2

Release 2.9.3-beta2

sergiou87 committed:
Bump version and changelog to 2.9.3

sergiou87 committed:
Use the right icon to _unfold_ code hunks

sergiou87 committed:
Use empty lines for the dummy hunk

sergiou87 committed:
Merge branch 'development' into release/2.9.3

haykam821 committed:
Fix the menu item to view a committed file change on GitHub not being disabled for unpushed commits

haykam821 committed:
Merge branch 'development' into view-committed-file-change-on-github

niik committed:
Use same width for hunk expansion handles as we do for line numbers

sergiou87 committed:
Merge pull request #12887 from desktop/contain-yourself

Use same width for hunk expansion handles as we do for line numbers

sergiou87 committed:
Merge pull request #12882 from desktop/diff-expansion-improvements

Small fixes/improvements in diff expansion

tidy-dev committed:
Don't update multi commit operation if it is already complete

sergiou87 committed:
Merge pull request #12877 from desktop/release/2.9.3

Release 2.9.3

haykam821 committed:
Simplify boolean logic for determining whether a selected commit is local

sergiou87 committed:
Use file contents after splitting into lines for diffs

sergiou87 committed:
Keep file contents in memory for diff expansion

sergiou87 committed:
Merge pull request #12492 from haykam821/view-committed-file-change-on-github

Add a menu item to view a committed file change on GitHub

tidy-dev committed:
Merge pull request #12888 from desktop/fix-merge-conflict-regression

Don't update multi commit operation if it is already complete

tidy-dev committed:
Merge pull request #12833 from desktop/high-contrast

Customizable high contrast theme

dependabot[bot] committed:
Bump tar from 4.4.15 to 4.4.19 in /app

Bumps [tar](https://github.com/npm/node-tar) from 4.4.15 to 4.4.19.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v4.4.15...v4.4.19)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

tidy-dev committed:
Release 2.9.4-beta1

sergiou87 committed:
Clean up some conditionals

sergiou87 committed:
Cache file contents in memory for side-by-side diffs too

tidy-dev committed:
Update changelog.json

Co-authored-by: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Update changelog.json

Commit:
Mention host key verification as possible auth issue

sergiou87 committed:
Refactor to use `IFileContents` instead of a new interface

tidy-dev committed:
Merge pull request #12892 from desktop/releases/2.9.4-beta1

Release 2.9.4-beta1

Commit:
Merge remote-tracking branch 'upstream/development' into codemirror-5.60+-upgrade

sergiou87 committed:
Merge pull request #11950 from say25/codemirror-5.60+-upgrade

Upgrade to CodeMirror ^5.60.0

haykam821 committed:
Fix gaps in the commit message container revealing a different background

Fixes #12900

Commit:
Update error messages

sergiou87 committed:
Load file contents for text diffs before rendering the diff

sergiou87 committed:
Enable diff loading optimizations for side-by-side diffs

sergiou87 committed:
Fix UI glitch with bottom dummy hunk

sergiou87 committed:
Merge pull request #12893 from desktop/diff-expansion-optimizations

Diff expansion optimizations

sergiou87 committed:
Fix non-text diffs (including large text diffs)

sergiou87 committed:
Only pass file contents down if the file can be expanded

Commit:
Support host authenticity of newer OpenSSH releases

Commit:
Show ssh key type of server in popup

niik committed:
Square bottom edges when co-authors is showing

niik committed:
Merge pull request #12901 from haykam821/fix-commit-message-container-gaps

Fix gaps in the commit message container revealing a different background

sergiou87 committed:
Merge pull request #12894 from thsmdt/development

Add host key verification to auth issue list

niik committed:
Update roadmap.md

niik committed:
Update CONTRIBUTING.md

sergiou87 committed:
Improve comments around the new `fileContents` prop

sergiou87 committed:
Merge pull request #12917 from desktop/doxipoxy

Update roadmap and contributing docs

sergiou87 committed:
Clean up some `setState` calls

sergiou87 committed:
Decouple syntax highlighting from diff expansion

sergiou87 committed:
Remove unnecessary usages of diff expansion feature flag

sergiou87 committed:
Refactor "can expand diff" logic a bit more

sergiou87 committed:
Re-enable syntax highlighting for large text diffs

sergiou87 committed:
Add some unit tests to ensure parsing add ssh host prompt doesn't break

sergiou87 committed:
Merge pull request #12919 from desktop/test-ssh-prompt-parsing

Add unit tests for our SSH host prompt parser

sergiou87 committed:
Highlight syntax again when file contents change

Co-Authored-By: Markus Olsson <634063+niik@users.noreply.github.com>

sergiou87 committed:
Merge pull request #12874 from desktop/win32-editor-improvements

Windows editor integration improvements

niik committed:
Merge pull request #12880 from desktop/dependabot/npm_and_yarn/app/tar-4.4.19

Bump tar from 4.4.15 to 4.4.19 in /app

sergiou87 committed:
Merge pull request #12903 from desktop/load-diff-at-once

Improve diff render and load

Commit:
Revise error message for failure to delete repository directory

tidy-dev committed:
Merge pull request #12902 from patrykmichalik/development

Update error messages

tidy-dev committed:
Check for merge head too

Move check lower and documentation

tidy-dev committed:
Merge pull request #12923 from desktop/merge-detection-regression

Fix merge head check regression introduced with squash merge

niik committed:
Always show the add repo dialog when adding from command line

niik committed:
:art: cleanup

niik committed:
:fire: unused import

niik committed:
Merge pull request #12935 from desktop/addask

Show add repo dialog when opening repo from CLI

tidy-dev committed:
Feature flag for CI check runs

tidy-dev committed:
Check List Item

tidy-dev committed:
Add the kind of check it is to the models

tidy-dev committed:
Add a check list component to display in popup

tidy-dev committed:
Add css to ui css list

tidy-dev committed:
Show popup with list

tidy-dev committed:
Woops add that back in

tidy-dev committed:
Css symbol alignment with check run name

tidy-dev committed:
meh

tidy-dev committed:
Add high-contrast mixin

tidy-dev committed:
Move high-contrast theme to high-contrast variable file

Two benefits:
- Clear definition of styles
- If we remove customization it high-contrast can be stand alone

tidy-dev committed:
Fix unit test

tidy-dev committed:
Build a short description

tidy-dev committed:
Build short description for check runs

tidy-dev committed:
Focus in the hc variables

fonsp committed:
run prettier

theodore-s-beers committed:
Add support for WezTerm on macOS

niik committed:
Merge pull request #12497 from fonsp/patch-1

tidy-dev committed:
Remove important

tidy-dev committed:
Merge pull request #12952 from desktop/stop-the-bleed

Prevent styles from high-contrast theme from leaking

niik committed:
Merge pull request #12957 from theodore-s-beers/wezterm

tidy-dev committed:
Release 2.9.4-beta1

tidy-dev committed:
Update changelog.json

Co-authored-by: Markus Olsson <j.markus.olsson@gmail.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Markus Olsson <j.markus.olsson@gmail.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Markus Olsson <j.markus.olsson@gmail.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Markus Olsson <j.markus.olsson@gmail.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

tidy-dev committed:
Update changelog.json

Co-authored-by: Billy Griffin <5091167+billygriffin@users.noreply.github.com>

niik committed:
Help function to compare diffs

niik committed:
Store both the diff and the file when loading

We can't determine if we need to reload the file contents without knowing if the diff has changed

niik committed:
We can't reuse the file contents and diff if the diff has changed

niik committed:
Whoops

niik committed:
:book:

niik committed:
Use the already captured variable

niik committed:
:book:

sergiou87 committed:
Only consider invalid tokens when 401s come from GH(E)

sergiou87 committed:
Don't notify invalidated token for auth-related requests

Co-Authored-By: Markus Olsson <634063+niik@users.noreply.github.com>

niik committed:
Merge pull request #12964 from desktop/the-matrix-reloaded-except-its-a-diff

Reload diff when app regains focus

sergiou87 committed:
Merge pull request #12965 from desktop/invalid-token-fixes

Only consider invalid tokens when 401s come from GH(E)

tidy-dev committed:
Merge pull request #12960 from desktop/releases/2.9.4-beta2

Release 2.9.4-beta2

sergiou87 committed:
Bump Electron to v11.5.0

sergiou87 committed:
Disable `setAlmostImmediate`

tidy-dev committed:
Merge branch 'generate-descriptions' into display-checks

tidy-dev committed:
Improved styles

tidy-dev committed:
Rename to check run for api searchability

remove old

tidy-dev committed:
Rename check run list item for better searchability

Fix prop

tidy-dev committed:
Remove console log

tidy-dev committed:
Clearer naming

dependabot[bot] committed:
Bump tmpl from 1.0.4 to 1.0.5

Bumps [tmpl](https://github.com/daaku/nodejs-tmpl) from 1.0.4 to 1.0.5.
- [Release notes](https://github.com/daaku/nodejs-tmpl/releases)
- [Commits](https://github.com/daaku/nodejs-tmpl/commits/v1.0.5)

---
updated-dependencies:
- dependency-name: tmpl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

niik committed:
Fix mismatched branches

https://github.com/github/codeql-action/blob/7112cdaa06f3b661965fe3e2e93c9acaadbe85f3/src/actions-util.ts#L182

niik committed:
Tighten up permissions

https://github.com/github/codeql-action#usage

niik committed:
Remove unnecessary head checkout

See https://github.com/github/codeql-action/blob/7112cdaa06f3b661965fe3e2e93c9acaadbe85f3/src/actions-util.ts#L186

niik committed:
:art: Update comments from template

niik committed:
Read-only GITHUB_TOKEN for CI jobs

sergiou87 committed:
Bump dugite to v2.32.0-1 and fix `remote` tests

sergiou87 committed:
Remove worktree-related code

sergiou87 committed:
Make checkout-index to fail silently on error 1

Error code 1 started being returned in https://github.com/git/git/commit/7e41061588eab25d0c015e79aa8d04956bee482a

sergiou87 committed:
Merge pull request #12997 from desktop/dependabot/npm_and_yarn/tmpl-1.0.5

Bump tmpl from 1.0.4 to 1.0.5

sergiou87 committed:
Merge pull request #12991 from desktop/bump-electron-v11-5-0

Bump Electron to v11.5.0

sergiou87 committed:
Merge pull request #13000 from desktop/bump-dugite-v2.32.0-1

Bump dugite to v2.32.0-1

sergiou87 committed:
Bump version and changelog to 2.9.4-beta3

sergiou87 committed:
Delete WebView2Loader.dll from Windows builds

That file cannot be signed and makes our builds fail. More info: https://github.com/microsoft/Git-Credential-Manager-Core/issues/462

tidy-dev committed:
Remove duplicate check

Co-authored-by: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Use optional chaining

Co-authored-by: Sergio Padrino <sergio.padrino@gmail.com>

tidy-dev committed:
Fix caret position

tidy-dev committed:
No flicker on second badge click

tidy-dev committed:
Close branch foldout on popover open

tidy-dev committed:
Center text - Adjust height of description so that it matches height of text

tidy-dev committed:
Drop status symbol a pixel

niik committed:
Initialize diff syntax mode when file contents change

niik committed:
Init diff syntax mode as soon as we've got an editor

niik committed:
Merge pull request #12998 from desktop/workflow-cleanup

Update CodeQL workflow using best practices and reduce token permissions

niik committed:
Merge pull request #13004 from desktop/initial-unified-highlight

Fix missing initial syntax highlighting for simple unified diffs

sergiou87 committed:
Merge pull request #13001 from desktop/releases/2.9.4-beta3

Release 2.9.4-beta3

sergiou87 committed:
Scroll diff to top if we switched to a new file

sergiou87 committed:
Merge branch 'development' into display-checks

sergiou87 committed:
Merge pull request #13010 from desktop/scroll-to-top

Scroll diff to top if we switched to a new file

tidy-dev committed:
Better positioning - doesn't have weird space on fullscreen

tidy-dev committed:
Merge pull request #12944 from desktop/display-checks

Display CI checks

alexanderadam committed:
feat: add syntax mapping for HAML - feature #13009

niik committed:
Merge pull request #13015 from alexanderadam/features/add_syntax_mapping_for_HAML_issue_13009

add syntax mapping for HAML

ADustyOldMuffin committed:
Add PR suggestions.

ADustyOldMuffin committed:
Add PR suggestions.

ADustyOldMuffin commented:
> > I think there is another error or something that just shows a list of files below the message, but I can't find it 🤔
> 
> @ADustyOldMuffin I found it! It's the confirmation dialog that shows up when you select multiple files in the `Changes` tab, right-click them and select `Discard changes`
> 
> ![image](https://user-images.githubusercontent.com/1083228/130414581-edf03945-1f7a-490d-93d8-866697b7d02c.png)
> 
> You will find it in `discard-changes-dialog.tsx`, and key is in the `PathText` component: https://github.com/desktop/desktop/blob/f09e9ec53e771aa90787ff8e8cb058bf6437f5bc/app/src/ui/discard-changes/discard-changes-dialog.tsx/#L128-L151

Sorry for the super late reply! (Life) I'm curious on any tips as to how I might go about doing something along those lines since this is returned as just text. I'll continue to look into something of this nature.

ADustyOldMuffin committed:
Update logic/regex to include filesize

ADustyOldMuffin committed:
Update logic/regex to include filesize

ADustyOldMuffin committed:
Fix some merge error.

ADustyOldMuffin committed:
Merge branch 'give-file-in-error-message' of https://github.com/ADustyOldMuffin/desktop into give-file-in-error-message

ADustyOldMuffin commented:
Also in regards to your want of the file size I've added it, I also ran into some odd behavior in that the error message returned from git only ever returns 1 offending file (even if there is more than 1).

![image](https://user-images.githubusercontent.com/41391321/134845235-ba538d15-af9e-438c-a297-88c2422c868d.png)

ADustyOldMuffin commented:
Since I took to long my candy is gone. I'll be making a new PR since I seem to have messed up this one.
